### ISSUE TITLE
Issues with http manage client connection sharing

### ISSUE THREAD

**Issue by derms** [marklogic-community/ml-app-deployer#244]
Using version 3.4.0 of ml-app-deployer , I am starting to see redeploys failing with the following error - 

```
18:07:40.823 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * What went wrong:
18:07:40.823 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] Execution failed for task ':mlDeployApp'.
18:07:40.823 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] > I/O error on GET request for "http://localhost:8002/manage/v2/databases": null; nested exception is org.apache.http.client.ClientProtocolException
```

Full (relevant) dump -

```
18:07:34.761 [DEBUG] [org.apache.http.client.protocol.RequestAuthCache] Auth cache not set in the context
18:07:34.761 [DEBUG] [org.apache.http.impl.conn.PoolingHttpClientConnectionManager] Connection request: [route: {}->http://localhost:8002][total kept alive: 1; route allocated: 1 of 2; total allocated: 1 of 20]
18:07:34.761 [DEBUG] [org.apache.http.impl.conn.PoolingHttpClientConnectionManager] Connection leased: [id: 84][route: {}->http://localhost:8002][total kept alive: 0; route allocated: 1 of 2;total allocated: 1 of 20]
18:07:34.761 [DEBUG] [org.apache.http.impl.execchain.MainClientExec] Stale connection check
18:07:34.763 [DEBUG] [org.apache.http.impl.execchain.MainClientExec] Executing request PUT /manage/v2/protected-paths/12419341783135600661/properties HTTP/1.1
18:07:34.763 [DEBUG] [org.apache.http.impl.execchain.MainClientExec] Target auth state: UNCHALLENGED
18:07:34.763 [DEBUG] [org.apache.http.impl.execchain.MainClientExec] Proxy auth state: UNCHALLENGED
18:07:34.786 [DEBUG] [org.apache.http.impl.execchain.MainClientExec] Connection can be kept alive for 5000 MILLISECONDS
18:07:34.786 [DEBUG] [org.apache.http.impl.conn.PoolingHttpClientConnectionManager] Connection [id: 84][route: {}->http://localhost:8002] can be kept alive for 5.0 seconds
18:07:34.787 [DEBUG] [org.apache.http.impl.conn.PoolingHttpClientConnectionManager] Connection released: [id: 84][route: {}->http://localhost:8002][total kept alive: 1; route allocated: 1 of 2; total allocated: 1 of 20]
18:07:34.787 [DEBUG] [org.springframework.web.client.RestTemplate] PUT request for "http://localhost:8002/manage/v2/protected-paths/12419341783135600661/properties" resulted in 204 (No Content)
18:07:34.787 [INFO] [com.marklogic.mgmt.resource.security.ProtectedPathManager] Updated protected-path at /manage/v2/protected-paths/12419341783135600661/properties
18:07:34.787 [INFO] [com.marklogic.appdeployer.impl.SimpleAppDeployer] Finished executing command [com.marklogic.appdeployer.command.security.DeployProtectedPathsCommand]

18:07:34.787 [INFO] [com.marklogic.appdeployer.impl.SimpleAppDeployer] Executing command [com.marklogic.appdeployer.command.security.DeployQueryRolesetsCommand] with sort order [97]
18:07:34.787 [INFO] [com.marklogic.appdeployer.command.security.DeployQueryRolesetsCommand] No resource directory found at: /Users/dsmyth/dev/workspace/malakoff-mederic-mdm/marklogic/user-config/security/query-rolesets
18:07:34.787 [INFO] [com.marklogic.appdeployer.impl.SimpleAppDeployer] Finished executing command [com.marklogic.appdeployer.command.security.DeployQueryRolesetsCommand]

18:07:34.787 [INFO] [com.marklogic.appdeployer.impl.SimpleAppDeployer] Executing command [com.marklogic.hub.deploy.commands.DeployHubTriggersDatabaseCommand] with sort order [100]
18:07:34.787 [INFO] [com.marklogic.hub.deploy.commands.DeployHubTriggersDatabaseCommand] Merging JSON files at locations: [/Users/dsmyth/dev/workspace/malakoff-mederic-mdm/marklogic/hub-internal-config/databases/triggers-database.json]
18:07:34.788 [INFO] [com.marklogic.mgmt.ManageClient] Sending XML GET request as user 'admin' to path: /manage/v2/databases
18:07:34.788 [DEBUG] [org.springframework.web.client.RestTemplate] Created GET request for "http://localhost:8002/manage/v2/databases"
18:07:34.788 [DEBUG] [org.springframework.web.client.RestTemplate] Setting request Accept header to [text/plain, application/json, application/*+json, */*]
18:07:34.788 [DEBUG] [org.apache.http.client.protocol.RequestAddCookies] CookieSpec selected: best-match
18:07:34.788 [DEBUG] [org.apache.http.client.protocol.RequestAddCookies] Cookie [version: 0][name: TxnID][value: null][domain: localhost][path: /][expiry: null] match [localhost:8002/manage/v2/databases]
18:07:34.788 [DEBUG] [org.apache.http.client.protocol.RequestAuthCache] Auth cache not set in the context
18:07:34.788 [DEBUG] [org.apache.http.impl.conn.PoolingHttpClientConnectionManager] Connection request: [route: {}->http://localhost:8002][total kept alive: 1; route allocated: 1 of 2; total allocated: 1 of 20]
18:07:34.788 [DEBUG] [org.apache.http.impl.conn.PoolingHttpClientConnectionManager] Connection leased: [id: 84][route: {}->http://localhost:8002][total kept alive: 0; route allocated: 1 of 2;total allocated: 1 of 20]
18:07:34.788 [DEBUG] [org.apache.http.impl.execchain.MainClientExec] Stale connection check
18:07:34.790 [DEBUG] [org.apache.http.impl.execchain.MainClientExec] Executing request GET /manage/v2/databases HTTP/1.1
18:07:34.790 [DEBUG] [org.apache.http.impl.execchain.MainClientExec] Target auth state: UNCHALLENGED
18:07:34.790 [DEBUG] [org.apache.http.impl.execchain.MainClientExec] Proxy auth state: UNCHALLENGED
18:07:34.807 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response: 12419341783135600661HTTP/1.1 200 OK
18:07:34.807 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response: Content-type: application/xml; charset=UTF-8
18:07:34.807 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response: Cache-Control: no-cache
18:07:34.807 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response: Expires: -1
18:07:34.807 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response: Server: MarkLogic
18:07:34.807 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response: Content-Length: 4830
18:07:34.807 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response: Connection: Keep-Alive
18:07:34.807 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response: Keep-Alive: timeout=5
18:07:34.807 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:
18:07:34.807 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response: <database-default-list xsi:schemaLocation="http://marklogic.com/manage/databases manage-databases.xsd" xmlns="http://marklogic.com/manage/databases" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
18:07:34.807 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:   <meta>
18:07:34.807 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     <uri>/manage/v2/databases</uri>
18:07:34.807 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     <current-time>2018-02-14T17:07:34.792379Z</current-time>
18:07:34.807 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     <elapsed-time units="sec">0.01374</elapsed-time>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:   </meta>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:   <list-items>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     <list-count units="quantity">17</list-count>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     <list-item array="true">
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <uriref>/manage/v2/databases/App-Services</uriref>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <idref>6872298475521566210</idref>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <nameref>App-Services</nameref>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     </list-item>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     <list-item array="true">
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <uriref>/manage/v2/databases/Documents</uriref>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <idref>14360075087226493783</idref>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <nameref>Documents</nameref>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     </list-item>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     <list-item array="true">
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <uriref>/manage/v2/databases/Extensions</uriref>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <idref>16928318436554712183</idref>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <nameref>Extensions</nameref>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     </list-item>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     <list-item array="true">
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <uriref>/manage/v2/databases/Fab</uriref>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <idref>4606600281531119809</idref>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <nameref>Fab</nameref>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     </list-item>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     <list-item array="true">
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <uriref>/manage/v2/databases/Last-Login</uriref>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <idref>16647433068632865879</idref>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <nameref>Last-Login</nameref>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     </list-item>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     <list-item array="true">
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <uriref>/manage/v2/databases/Meters</uriref>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <idref>9097928595410976725</idref>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <nameref>Meters</nameref>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     </list-item>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     <list-item array="true">
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <uriref>/manage/v2/databases/Modules</uriref>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <idref>9291645577185973002</idref>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <nameref>Modules</nameref>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     </list-item>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     <list-item array="true">
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <uriref>/manage/v2/databases/RCU-FINAL</uriref>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <idref>11041639511375699110</idref>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <nameref>RCU-FINAL</nameref>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     </list-item>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     <list-item array="true">
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <uriref>/manage/v2/databases/RCU-JOBS</uriref>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <idref>17454370164825631039</idref>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <nameref>RCU-JOBS</nameref>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     </list-item>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     <list-item array="true">
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <uriref>/manage/v2/databases/RCU-MODULES</uriref>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <idref>6005699396006682565</idref>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <nameref>RCU-MODULES</nameref>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     </list-item>
18:07:34.808 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     <list-item array="true">
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <uriref>/manage/v2/databases/RCU-SCHEMAS</uriref>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <idref>5481491213631766276</idref>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <nameref>RCU-SCHEMAS</nameref>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     </list-item>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     <list-item array="true">
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <uriref>/manage/v2/databases/RCU-STAGING</uriref>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <idref>7053909562721913167</idref>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <nameref>RCU-STAGING</nameref>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     </list-item>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     <list-item array="true">
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <uriref>/manage/v2/databases/RCU-TRACING</uriref>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <idref>2626045066193462947</idref>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <nameref>RCU-TRACING</nameref>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     </list-item>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     <list-item array="true">
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <uriref>/manage/v2/databases/RCU-TRIGGERS</uriref>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <idref>11772540032500550941</idref>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <nameref>RCU-TRIGGERS</nameref>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     </list-item>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     <list-item array="true">
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <uriref>/manage/v2/databases/Schemas</uriref>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <idref>10385013348161268023</idref>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <nameref>Schemas</nameref>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     </list-item>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     <list-item array="true">
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <uriref>/manage/v2/databases/Security</uriref>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <idref>2457355379337248258</idref>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <nameref>Security</nameref>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     </list-item>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     <list-item array="true">
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <uriref>/manage/v2/databases/Triggers</uriref>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <idref>14013875757041511784</idref>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <nameref>Triggers</nameref>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     </list-item>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:   </list-items>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:   <related-views>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     <related-view array="true">
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <view-type>root</view-type>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <view-name>default</view-name>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <view-uri>/manage/v2</view-uri>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     </related-view>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     <related-view array="true">
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <view-type>list</view-type>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <view-name>metrics</view-name>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <view-uri>/manage/v2/databases?view=metrics</view-uri>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     </related-view>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     <related-view array="true">
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <view-type>list</view-type>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <view-name>package</view-name>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <view-uri>/manage/v2/databases?view=package</view-uri>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     </related-view>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     <related-view array="true">
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <view-type>list</view-type>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <view-name>properties-schema</view-name>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <view-uri>/manage/v2/databases?view=properties-schema</view-uri>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     </related-view>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     <related-view array="true">
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <view-type>list</view-type>
18:07:34.809 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <view-name>schema</view-name>
18:07:34.810 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <view-uri>/manage/v2/databases?view=schema</view-uri>
18:07:34.810 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     </related-view>
18:07:34.810 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     <related-view array="true">
18:07:34.810 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <view-type>list</view-type>
18:07:34.810 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <view-name>health</view-name>
18:07:34.810 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <view-uri>/manage/v2?view=health&amp;health-resource=database</view-uri>
18:07:34.810 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     </related-view>
18:07:34.810 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     <related-view array="true">
18:07:34.810 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <view-type>list</view-type>
18:07:34.810 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <view-name>docs</view-name>
18:07:34.810 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:       <view-uri>http://docs.marklogic.com/REST/management/databases</view-uri>
18:07:34.810 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:     </related-view>
18:07:34.810 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response:   </related-views>
18:07:34.810 [DEBUG] [org.apache.http.impl.conn.DefaultHttpResponseParser] Garbage in response: </database-default-list>
18:07:36.863 [null] [org.gradle.process.internal.health.memory.MemoryManager]
18:07:36.863 [DEBUG] [org.gradle.process.internal.health.memory.MemoryManager] Emitting OS memory status event {Total: 17179869184, Free: 6860365824}
18:07:36.863 [DEBUG] [org.gradle.launcher.daemon.server.health.LowMemoryDaemonExpirationStrategy] Received memory status update: {Total: 17179869184, Free: 6860365824}
18:07:36.863 [DEBUG] [org.gradle.process.internal.health.memory.MemoryManager] Emitting JVM memory status event {Maximum: 954728448, Committed: 720896000}
18:07:34.867 [null] [org.gradle.internal.progress.DefaultBuildOperationExecutor]
> Task :mlDeployApp FAILED
18:07:40.818 [DEBUG] [org.apache.http.impl.conn.DefaultManagedHttpClientConnection] http-outgoing-84: Close connection
18:07:40.818 [DEBUG] [org.apache.http.impl.conn.DefaultManagedHttpClientConnection] http-outgoing-84: Shutdown connection
18:07:40.818 [DEBUG] [org.apache.http.impl.execchain.MainClientExec] Connection discarded
18:07:40.818 [DEBUG] [org.apache.http.impl.conn.DefaultManagedHttpClientConnection] http-outgoing-84: Close connection
18:07:40.818 [DEBUG] [org.apache.http.impl.conn.PoolingHttpClientConnectionManager] Connection released: [id: 84][route: {}->http://localhost:8002][total kept alive: 0; route allocated: 0 of 2; total allocated: 0 of 20]
18:07:40.818 [DEBUG] [org.gradle.internal.progress.DefaultBuildOperationExecutor] Completing Build operation 'Execute deployApp for :mlDeployApp'
18:07:40.818 [DEBUG] [org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter] Removed task artifact state for {} from context.
18:07:40.818 [DEBUG] [org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter] Finished executing task ':mlDeployApp'
18:07:40.818 [DEBUG] [org.gradle.internal.progress.DefaultBuildOperationExecutor] Completing Build operation 'Task :mlDeployApp'
18:07:40.818 [null] [org.gradle.internal.progress.DefaultBuildOperationExecutor]
18:07:40.818 [DEBUG] [org.gradle.internal.progress.DefaultBuildOperationExecutor] Build operation 'Task :mlDeployApp' completed
18:07:40.818 [INFO] [org.gradle.execution.taskgraph.DefaultTaskPlanExecutor] :mlDeployApp (Thread[Task worker for ':',5,main]) completed. Took 31.137 secs.
18:07:40.818 [DEBUG] [org.gradle.internal.work.DefaultWorkerLeaseService] Worker lease root.1.12 completed (1 worker(s) in use)
18:07:40.818 [DEBUG] [org.gradle.internal.resources.AbstractTrackedResourceLock] Task worker for ':': released lock on root.1.12
18:07:40.818 [DEBUG] [org.gradle.internal.resources.AbstractTrackedResourceLock] Task worker for ':': released lock on :
18:07:40.818 [DEBUG] [org.gradle.execution.taskgraph.DefaultTaskPlanExecutor] Task worker [Thread[Task worker for ':',5,main]] finished, busy: 32.308 secs, idle: 0.005 secs
18:07:40.819 [DEBUG] [org.gradle.execution.taskgraph.DefaultTaskPlanExecutor] Task worker [Thread[Task worker for ':' Thread 2,5,main]] finished, busy: 0.0 secs, idle: 32.311 secs
18:07:40.820 [DEBUG] [org.gradle.execution.taskgraph.DefaultTaskPlanExecutor] Task worker [Thread[Task worker for ':' Thread 3,5,main]] finished, busy: 0.0 secs, idle: 32.311 secs
18:07:40.820 [DEBUG] [org.gradle.execution.taskgraph.DefaultTaskPlanExecutor] Task worker [Thread[Daemon worker Thread 3,5,main]] finished, busy: 0.0 secs, idle: 32.311 secs
18:07:40.819 [DEBUG] [org.gradle.execution.taskgraph.DefaultTaskPlanExecutor] Task worker [Thread[Task worker for ':' Thread 6,5,main]] finished, busy: 0.0 secs, idle: 32.31 secs
18:07:40.819 [DEBUG] [org.gradle.execution.taskgraph.DefaultTaskPlanExecutor] Task worker [Thread[Task worker for ':' Thread 5,5,main]] finished, busy: 0.0 secs, idle: 32.311 secs
18:07:40.820 [DEBUG] [org.gradle.execution.taskgraph.DefaultTaskPlanExecutor] Task worker [Thread[Task worker for ':' Thread 7,5,main]] finished, busy: 0.0 secs, idle: 32.312 secs
18:07:40.820 [DEBUG] [org.gradle.execution.taskgraph.DefaultTaskPlanExecutor] Task worker [Thread[Task worker for ':' Thread 4,5,main]] finished, busy: 0.0 secs, idle: 32.311 secs
18:07:40.822 [DEBUG] [org.gradle.internal.progress.DefaultBuildOperationExecutor] Completing Build operation 'Run tasks'
18:07:40.823 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]
18:07:40.823 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] FAILURE: Build failed with an exception.
18:07:40.823 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]
18:07:40.823 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * What went wrong:
18:07:40.823 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] Execution failed for task ':mlDeployApp'.
18:07:40.823 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] > I/O error on GET request for "http://localhost:8002/manage/v2/databases": null; nested exception is org.apache.http.client.ClientProtocolException
18:07:40.823 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]
18:07:40.823 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * Try:
18:07:40.823 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] Run with --stacktrace option to get the stack trace.  Run with --scan to get full insights.
18:07:40.823 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]
18:07:40.823 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * Get more help at https://help.gradle.org
18:07:40.823 [ERROR] [org.gradle.internal.buildevents.BuildResultLogger]
18:07:40.823 [ERROR] [org.gradle.internal.buildevents.BuildResultLogger] BUILD FAILED in 34s
```

If I change the com.marklogic.rest.util.RestTemplateUtil.newRestTemplate() method to include -

```
		httpClientBuilder.setConnectionReuseStrategy(new NoConnectionReuseStrategy());
```

then everything works fine.

I will create a pull request for this fix

**Comment by rjrudin** [comment]
RestTemplateUtil changed in 3.4.0, but I can't see how those changes would impact connectivity. Do you have SSL enabled on the connection to 8002?

Also, what version of ML is this?

I'll get read up on the pros/cons of using the NoConnection strategy.

**Comment by derms** [comment]
ML version is 9.0-3.1
No SSL - just standard DIGEST on 8002. 

Btw is might be worth looking at adding useSystemProperties() prior to calling build too - 
http://hc.apache.org/httpcomponents-client-ga/httpclient/apidocs/org/apache/http/impl/client/HttpClientBuilder.html#useSystemProperties() 

could then do system prop overrides as required too.

**Comment by rjrudin** [comment]
Addressed via pull request #246
